### ISSUE TITLE
Delete with no revisions present throws

### ISSUE THREAD

**Issue by evfool** [javers/javers#1108]
**Clear description of my expectations versus reality**
Deleting an object with no history stored throws an exception:
org.javers.common.exception.JaversException: CANT_DELETE_OBJECT_NOT_FOUND: failed to delete object XXXXXXX, it doesn't exists in JaversRepository

This looks like the issue reported in #461 and fixed with merge of https://github.com/javers/javers/pull/477, the test is still there, but an exception is still being thrown.

**Javers' Version**
6.0.1

**Additional context**
Full stacktrace:

```
org.javers.common.exception.JaversException: CANT_DELETE_OBJECT_NOT_FOUND: failed to delete object XXXXXXX, it doesn't exists in JaversRepository
at org.javers.core.commit.CommitFactory.lambda$createTerminalByGlobalId$1(CommitFactory.java:58)
at java.util.Optional.orElseThrow(Optional.java:290)
at org.javers.core.commit.CommitFactory.createTerminalByGlobalId(CommitFactory.java:58)
at org.javers.core.commit.CommitFactory.createTerminal(CommitFactory.java:66)
at org.javers.core.JaversCore.commitShallowDelete(JaversCore.java:150)
at org.javers.spring.jpa.JaversTransactionalDecorator.commitShallowDelete(JaversTransactionalDecorator.java:95)
at org.javers.spring.jpa.JaversTransactionalDecorator$$FastClassBySpringCGLIB$$acb40bd0.invoke(<generated>)
at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:779)
at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:750)
at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:123)
at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:388)
at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:119)
at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:750)
at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:692)
at org.javers.spring.jpa.JaversTransactionalDecorator$$EnhancerBySpringCGLIB$$7f93ca29.commitShallowDelete(<generated>)
at org.javers.spring.auditable.aspect.JaversCommitAdvice.commitShallowDelete(JaversCommitAdvice.java:87)
at org.javers.spring.auditable.aspect.JaversCommitAdvice.commitDeleteMethodArguments(JaversCommitAdvice.java:62)
at org.javers.spring.auditable.aspect.JaversAuditableAspect.commitDeleteAdvice(JaversAuditableAspect.java:52)
at sun.reflect.GeneratedMethodAccessor277.invoke(Unknown Source)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.lang.reflect.Method.invoke(Method.java:498)
```

**Comment by bartoszwalacik** [comment]
I'm not convinced it's a bug. It's a missing feature:
Possibility to deleting an object from JaversRepository even if it doesn't exists in JaversRepository.

We will accept a PR.

**Comment by evfool** [comment]
Ok, this can be considered as a new feature. My first idea would be to allow creating a TERMINAL snapshot when there is no previous snapshot. Would this be acceptable from your point of view, or there should always be an INITIAL snapshot?

**Comment by bartoszwalacik** [comment]
it's ok, there will be only a terminal snapshot

**Comment by bartoszwalacik** [comment]
released in 6.2.2
